src/number_theory/modular_forms/jacobi_theta.lean

```diff
@@ -92,3 +96,89 @@ begin
     congr' 1,
     ring_nf }
 end
+
+lemma jacobi_theta_has_sum_nat (τ : ℍ) :
+  has_sum (λ (n : ℕ), cexp (π * I * (n + 1) ^ 2 * τ)) ((jacobi_theta τ - 1) / 2) :=
+begin
+  have := (jacobi_theta_summable τ.im_pos).has_sum.sum_nat_of_sum_int,
+  rw ←@has_sum_nat_add_iff' ℂ _ _ _ _ 1 at this,
+  simp_rw [finset.sum_range_one, int.cast_neg, int.cast_coe_nat, nat.cast_zero, neg_zero,
+    int.cast_zero, sq (0:ℂ), mul_zero, zero_mul, neg_sq, ←mul_two, complex.exp_zero,
+    add_sub_assoc, (by norm_num : (1 : ℂ) - 1 * 2 = -1), ←sub_eq_add_neg,
+    nat.cast_add, nat.cast_one] at this,
+  convert this.div_const 2,
+  simp_rw mul_div_cancel _ two_ne_zero,
+end
+
+lemma jacobi_theta_eq_tsum_nat (τ : ℍ) :
+  jacobi_theta τ = 1 + 2 * ∑' (n : ℕ), cexp (π * I * (n + 1) ^ 2 * τ) :=
+by rw [(jacobi_theta_has_sum_nat τ).tsum_eq, mul_div_cancel' _ (two_ne_zero' ℂ), ←add_sub_assoc,
+  add_sub_cancel']
+
+/-- An explicit upper bound for `‖jacobi_theta τ - 1‖`. -/
+lemma jacobi_theta_sub_one_norm_le (τ : ℍ) :
+  ‖jacobi_theta τ - 1‖ ≤ 2 / (1 - exp (-π * τ.im)) * exp (-π * τ.im) :=
+begin
+  suffices : ‖∑' (n : ℕ), cexp (π * I * (n + 1) ^ 2 * τ)‖ ≤ exp (-π * τ.im) / (1 - exp (-π * τ.im)),
+  { calc ‖jacobi_theta τ - 1‖ = 2 * ‖∑' (n : ℕ), cexp (π * I * (n + 1) ^ 2 * τ)‖ :
+      by rw [sub_eq_iff_eq_add'.mpr (jacobi_theta_eq_tsum_nat τ), norm_mul, complex.norm_eq_abs,
+        complex.abs_two]
+    ... ≤ 2 * (rexp (-π * τ.im) / (1 - rexp (-π * τ.im))) :
+      by rwa [mul_le_mul_left (zero_lt_two' ℝ)]
+    ... = 2 / (1 - rexp (-π * τ.im)) * rexp (-π * τ.im) : by rw [div_mul_comm, mul_comm] },
+  have : ∀ (n : ℕ), ‖cexp (π * I * (n + 1) ^ 2 * τ)‖ ≤ exp (-π * τ.im) ^ (n + 1),
+  { intro n,
```

Comment:
Isn't this the same argument as in the proof of `jacobi_theta_unif_summable`? Maybe you could extract a lemma saying that, if `R ≤ τ.im`, then `‖cexp (π * I * n ^ 2 * τ)‖ ≤ ...` and use it for both proofs.

Reply:
It seems I was too hasty in getting rid of `jacobi_theta_term_bound` in my last commit! I've reinstated it and used it here.